spring-ai-core/src/main/java/org/springframework/ai/model/function/AbstractToolCallSupport.java

```diff
@@ -149,6 +151,21 @@ protected List<ToolResponseMessage> executeFuncitons(AssistantMessage assistantM
 
 	}
 
-	abstract protected boolean isToolFunctionCall(TRes response);
+	protected boolean isToolCall(ChatResponse chatResponse, String toolCallFinishReason) {
+		Assert.hasText(toolCallFinishReason, "toolCallFinishReason cannot be null or empty");
+
+		if (chatResponse == null) {
+			return false;
+		}
+
+		var generations = chatResponse.getResults();
+		if (CollectionUtils.isEmpty(generations)) {
+			return false;
+		}
+
+		var generation = generations.get(0);
+		return !CollectionUtils.isEmpty(generation.getOutput().getToolCalls())
+				&& toolCallFinishReason.equalsIgnoreCase(generation.getMetadata().getFinishReason());
```

Comment:
This check could be improved further if we introduced some abstractions of the FinishReason, too